components/script/dom/htmliframeelement.rs

```diff
@@ -131,30 +140,70 @@ impl HTMLIFrameElement {
         let load_info = IFrameLoadInfo {
-            load_data: load_data,
             parent_pipeline_id: global_scope.pipeline_id(),
             frame_id: self.frame_id,
-            old_pipeline_id: old_pipeline_id,
             new_pipeline_id: new_pipeline_id,
-            sandbox: sandboxed,
             is_private: private_iframe,
             frame_type: frame_type,
             replace: replace,
         };
-        global_scope
-              .constellation_chan()
-              .send(ConstellationMsg::ScriptLoadedURLInIFrame(load_info))
-              .unwrap();
+
+        if load_data.as_ref().map_or(false, |d| d.url.as_str() == "about:blank") {
+            let (pipeline_sender, pipeline_receiver) = ipc::channel().unwrap();
+
+            global_scope
+                  .constellation_chan()
+                  .send(ConstellationMsg::ScriptLoadedAboutBlankInIFrame(load_info, pipeline_sender))
+                  .unwrap();
+
+            let new_layout_info = NewLayoutInfo {
+                parent_info: Some((global_scope.pipeline_id(), frame_type)),
+                new_pipeline_id: new_pipeline_id,
+                frame_id: self.frame_id,
+                load_data: load_data.unwrap(),
+                pipeline_port: pipeline_receiver,
+                content_process_shutdown_chan: None,
+                window_size: None,
+                layout_threads: PREFS.get("layout.threads").as_u64().expect("count") as usize,
+            };
+
+            ScriptThread::process_attach_layout(new_layout_info);
+        } else {
+            let load_info = IFrameLoadInfoWithData {
+                info: load_info,
+                load_data: load_data,
+                old_pipeline_id: old_pipeline_id,
+                sandbox: sandboxed,
+            };
+            global_scope
+                  .constellation_chan()
+                  .send(ConstellationMsg::ScriptLoadedURLInIFrame(load_info))
+                  .unwrap();
+        }
 
         if PREFS.is_mozbrowser_enabled() {
             // https://developer.mozilla.org/en-US/docs/Web/Events/mozbrowserloadstart
             self.dispatch_mozbrowser_event(MozBrowserEvent::LoadStart);
         }
     }
 
-    pub fn process_the_iframe_attributes(&self) {
+    /// https://html.spec.whatwg.org/multipage/#process-the-iframe-attributes
+    fn process_the_iframe_attributes(&self, mode: ProcessingMode) {
+        // TODO: srcdoc
+
+        // https://github.com/whatwg/html/issues/490
+        if mode == ProcessingMode::FirstTime && !self.upcast::<Element>().has_attribute(&local_name!("src")) {
+            let window = window_from_node(self);
+            let event_loop = window.dom_manipulation_task_source();
+            let _ = event_loop.queue(box IframeLoadEventSteps::new(self),
+                                     window.upcast());
+            return;
+        }
+
         let url = self.get_url();
 
+        // TODO: check ancestor browsing contexts for same URL
```

Comment:
Can you expand on this TODO a bit? I'm not sure what we'd do differently if we find the same URL in an ancestor.